packages/x-data-grid-premium/src/hooks/features/rowGrouping/createGroupingColDef.tsx

```diff
@@ -25,11 +28,24 @@ const GROUPING_COL_DEF_DEFAULT_PROPERTIES: Omit<GridColDef, 'field'> = {
   disableReorder: true,
 };
 
-const GROUPING_COL_DEF_FORCED_PROPERTIES: Pick<GridColDef, 'type' | 'editable' | 'groupable'> = {
+const GROUPING_COL_DEF_FORCED_PROPERTIES_DEFAULT: Pick<
+  GridColDef,
+  'type' | 'editable' | 'groupable'
+> = {
   editable: false,
   groupable: false,
 };
 
+const GROUPING_COL_DEF_FORCED_PROPERTIES_DATA_SOURCE: Pick<
+  GridColDef,
+  'type' | 'editable' | 'groupable' | 'filterable' | 'sortable' | 'aggregable'
+> = {
+  ...GROUPING_COL_DEF_FORCED_PROPERTIES_DEFAULT,
+  // TODO: Support these features on the grouping column(s)
```

Comment:
what is preventing you reseting the grid and making a new request with the new parameters?

Reply:
Thanks for the question, I wanted to drop a comment on this but forgot.

We need to comeup with some standard way to communicate with the backend server that a specific operation (sorting/filtering/etc) is requested on a certain grouping column. Currently, hardcoded field names are used for these columns (both in tree data and row grouping) that are more frontend specific.

Tree data grouping field: `__tree_data_group__`
Row grouping single grouping column: `__row_group_by_columns_group__`
Row grouping multiple grouping columns: `__row_group_by_columns_group_${field}__`

Currently when sorting or filtering, it would yield into:

```tsx
const filterModel = {
  items: [
    {
      field: '__row_group_by_columns_group_company__',
      value: 'John',
      value: 'contains',
    },
    {
      field: '__tree_data_group__',
      value: 'John',
      value: 'startsWith',
    },
  ],
};

const sortModel = [
  {
    field: '__row_group_by_columns_group__',
    sort: 'asc',
  },
  {
    field: '__tree_data_group__',
    sort: 'asc',
  },
];
```

We need to translate these values in to some generic readable values based on the current configuration before passing them on to the data source. For example when there is a specific filtering criteria defined for single column multiple row grouping, we need to communicate it to the backend, a possible field name to use for it could be `group_${criteriaField}`

That's the reason I wanted to handle it separately, to gather some community/internal feedback on the naming first and also because it'd not only cover the row grouping but also tree data.

I'll create a separate issue for this and handle it afterwards.

Reply:
thanks for all the details
yes, that is fine
let's do it afterwards